### ISSUE TITLE
type Array assignment false warning

### ISSUE THREAD

**Issue by rts-barry** [okbob/plpgsql_check#15]
It seems that plpgsql_check_function gives a warning where no casting issues are there.

```SQL
CREATE TYPE t AS (t TEXT);

CREATE FUNCTION f(OUT ta t[]) AS $$
DECLARE
	t t;
BEGIN
	t.t := 'abc';
	ta[1] := t;
END;
$$ LANGUAGE plpgsql;

SELECT f(); -- works ok
SELECT plpgsql_check_function('f()');
/*
"warning:42804:6:assignment:target variable has different type than expression result"
"Detail: assign "text" value to "t" variable"
"Hint: The input expression type does not have an assignment cast to the target type."
*/
```

**Comment by okbob** [comment]
Hi 
should be fixed by 280cdc4744919ac0e6d1d959def4e309c11cdb22 commit

**Comment by rts-barry** [comment]
Thanks for making the commit, it solved the issue.
